Sources/SwiftKueryORM/TableInfo.swift

```diff
@@ -35,18 +35,33 @@ public class TableInfo {
 
     func getInfo<T: Decodable>(_ idColumn: (name: String, type: SQLDataType.Type), _ tableName: String, _ type: T.Type) throws -> (info: TypeInfo, table: Table) {
         let typeString = "\(type)"
-        if let result = codableMap[typeString] {
+        var result: (TypeInfo, Table)? = nil
+        // Read from codableMap when no concurrent write is occuring
+        codableMapQueue.sync {
+            result = codableMap[typeString]
+        }
+        if let result = result {
             return result
         }
 
-        codableMapLock.wait()
-        defer { codableMapLock.signal() }
-
-        let typeInfo = try TypeDecoder.decode(type)
-        let result = (info: typeInfo, table: try constructTable(idColumn, tableName, typeInfo))
-        codableMap[typeString] = result
-
-        return result
+        var decodeError: Error?
+        // Prevent concurrent access to codableMap while writing
+        do {
```

Comment:
Is it necessary to capture decodeError? If you take the do/catch away, the error will be thrown out of queue.sync and this function will exit early (throwing).  The logic that follows could then be simplified to:
```swift
guard let decodeResult = result else {
  throw RequestError...
}
return decodeResult
```